docs/DeveloperGuide.adoc

```diff
@@ -0,0 +1,665 @@
+*** Pros: Easier to implement and improved code readability
+*** Cons: `ArrayList` will be filled up with unnecessary tasks that could have been removed. This might affect the
+time complexity of future addition or searching operations on the `ArrayList`.
+
+=== Repeat event feature
+This feature allow users to repeat their events, removing the need to insert the same event multiple times with different dates.
+
+==== Current Implementation
+The `RepeatCommand` class extends the `Command` class and either allows the stated event to repeat or to stop repeating.
+To allow an event to repeat, it will replace the current `Event` object with a `RepeatEvent` object (`RepeatEvent` inherits from
+ `Event`). Likewise, to stop repeating, it replaces the current `RepeatEvent` with a `Event` object. A detailed explanation and
+the difference between the 2 classes will be elaborated later.
+
+Given below is an example usage of the `repeat id/2 p/1w` command.
+
+*Step 1* +
+Method `parseCommand()` from the `Parser` class will be called to parse the command provided. Through this method, we will be able to obtain
+ information to get integers `eventID`, `numOfPeriod` and the string `typeOfPeriod`. +
+- `eventID` identifies the task that the user wishes to repeat. +
+- `numOfPeriod` and `typeOfPeriod` (`d`, `w`, `m`, or `y`) specifies how often the user wants to repeat the event.
+
+*Step 2* +
+After parsing, a new instance of RepeatCommand with `eventID`, `numOfPeriod` and `typeOfPeriod` initialized will be created.
+The `execute()` method of `RepeatCommand` will then be called.
+
+*Step 3* +
+The `execute()` method will check 3 things after it calls `getTask()` method from `TaskList` class to get the user input task.
+
+* It will check if the `eventID` provided refers to a valid `Event` task.
+* It will then check if `numOfPeriod` equals to 0. In which case, it will be setting the event to not repeat by calling `unsetRepeat`
+method from `RepeatCommand` class.
+** `unsetRepeat()` method will check if the given task is indeed a `RepeatEvent` object and then create a new `Event` object using the
+variables from `RepeatEvent` and replace it in the `TaskList`.
+* If it is not 0, it will set the event to repeating by calling `setRepeat()` method from the `RepeatCommand` class.
+** `setRepeat` method will use 2 of the variables (`numOfPeriod`, `typeOfPeriod`) to create a new `RepeatEvent` object and replace the
+current `Event` object at the `eventID` in `TaskList`.
+
+*Step 4* +
+After the `execute()` method completes, a new `CommandResult` class with a string containing the result of the execution.
+This string will be printed by calling `showToUser()` method in the `Ui` class. Then the event will be saved into local file by calling
+the `trySaveTaskList()` method from the `Atas` class.
+
+The following sequence diagram summarizes how repeat command operation works: +
+
+image::RepeatCommand_UML.png[Repeat Command Sequence Diagram]
```

Comment:
Clear sequence diagram on explaining this features ! 👍 
Possible minor fix: The arrow (circle in red) is disjoint from the repeatCommand box.
![Screenshot (9)_LI](https://user-images.githubusercontent.com/42569361/77922851-2b1a5280-72d4-11ea-973c-94ae23918571.jpg)